lighthouse-core/gather/driver.js

```diff
@@ -630,11 +630,11 @@ class Driver {
      * @param {() => void} resolve
      */
     function checkForQuiet(driver, resolve) {
-      if (cancelled) return;
+      if (cancelled) resolve();
```

Comment:
hang on might need to walk me through this, it was slightly different than what I was thinking :)

if it's been cancelled, we want to stop continuing and our calls to resolve don't really matter since they'll be ignored anyhow.

don't we want
```js
if (cancelled) return Promise.resolve()
```

seems like the current code will still kick off the `evaluateAsync` loop

Reply:
Oh yeah, good catch, I was confusing my `resolves` since it takes in the resolve, but in the cancelled case we want to return an empty resolved promise & return afaik, right?

Reply:
> return an empty resolved promise & return afaik

Yep, that checks out to me!